pyi.py

```diff
@@ -421,9 +421,54 @@ def _check_union_members(self, members: Sequence[ast.expr]) -> None:
         for member in members:
             members_by_dump.setdefault(ast.dump(member), []).append(member)
 
-        for members in members_by_dump.values():
-            if len(members) >= 2:
-                self.error(members[1], Y016.format(unparse(members[1])))
+        dupes_in_union = False
+        for member_list in members_by_dump.values():
+            if len(member_list) >= 2:
+                self.error(member_list[1], Y016.format(unparse(member_list[1])))
+                dupes_in_union = True
+
+        if not dupes_in_union:
+            self._check_for_multiple_literals(members)
```

Comment:
This would be slightly simpler with `return` after erroring.

Reply:
I agree, but I quite like that you get multiple errors if you have multiple dupes in one union? Otherwise it will tell you about the first duplicate, but won't tell you about the other duplicates (if you do have more than one duplicate).